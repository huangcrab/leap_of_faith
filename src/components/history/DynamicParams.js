import React from "react";
import styled from "styled-components";

const Quote = styled.blockquote`
  color: black;
  font-size: 12px;
  & span {
    font-weight: bold;
    color: #fd7f80;
  }
`;

export default function DynamicParams(props) {
  const { dynamic } = props;
  return (
    <Quote>
      {dynamic
        ? dynamic.map(item => {
            if (item.key !== "" && item.type !== "TABLE") {
              return (
                <div key={item.key}>
                  {item.key}:<span> {item.value}</span>
                </div>
              );
            } else return null;
          })
        : null}
    </Quote>
  );
}
