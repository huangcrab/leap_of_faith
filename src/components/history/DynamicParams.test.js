import React from "react";
import { shallow } from "enzyme";
import DynamicParams from "./DynamicParams";
describe("DynamicParams", () => {
  it("should render banner text correctly with given strings", () => {
    const objects = [
      { key: "one", value: "asd" },
      { key: "two", value: "dsa" }
    ];
    const component = shallow(<DynamicParams dynamic={objects} />);
    expect(component).toMatchSnapshot();
  });
});
