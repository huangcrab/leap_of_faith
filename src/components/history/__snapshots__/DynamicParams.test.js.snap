// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DynamicParams should render banner text correctly with given strings 1`] = `
<styled.blockquote>
  <div
    key="one"
  >
    one
    :
    <span>
       
      asd
    </span>
  </div>
  <div
    key="two"
  >
    two
    :
    <span>
       
      dsa
    </span>
  </div>
</styled.blockquote>
`;
